### PR TITLE
Add a test for multiple JsonQualifiers.

### PR DESCRIPTION
**Pull request by NightlyNexus**
And pass in an unmodifiable set to adapter factories.

### CODE REVIEW

File: tests/src/test/kotlin/se/ansman/kotshi/TestAdapterGeneration.kt [hunk 1]
```diff
@@ -156,4 +156,32 @@ class TestAdapterGeneration {
         assertEquals(GenericClassWithQualifier("Hello, world!"), actual)
         assertEquals(json, adapter.toJson(actual))
     }
-}
\ No newline at end of file
+
+    @Test
+    fun testMultipleJsonQualifiers() {
+        val adapter = Moshi.Builder().add(object : Any() {
```
Comment (ansman) [inline tests/src/test/kotlin/se/ansman/kotshi/TestAdapterGeneration.kt]: Could you wrap each statement on its own line.

Comment (NightlyNexus) [reply tests/src/test/kotlin/se/ansman/kotshi/TestAdapterGeneration.kt]: Yeah, happy to fix up the style (or when i next modify this file). Do you mean `add`, `build`, and `adapter` on new lines?

Comment (ansman) [reply tests/src/test/kotlin/se/ansman/kotshi/TestAdapterGeneration.kt]: Yeah, but I fixed it after I merged it so don't worry about it :)

Comment (NightlyNexus) [reply tests/src/test/kotlin/se/ansman/kotshi/TestAdapterGeneration.kt]: 👍 thanks!

**Pull request merged**